src/puppetlabs/trapperkeeper/services/status/status_core.clj

```diff
@@ -211,6 +211,37 @@
              (log/error e error-msg)
              (unknown-response (format "%s: %s" error-msg e))))))))
 
+(schema/defn ^:always-validate matching-service-info :- ServiceInfo
+  "Find a service info entry matching the service-status-version. If
+  service-status-version is nil the most recent service info is returned."
+  [service-name :- schema/Str
+   service :- [ServiceInfo]
+   service-status-version :- (schema/maybe schema/Int)]
```

Comment:
Unless there's a compelling reason not to, I think it's better to have two different arities of this function rather than take a final parameter that can be nil. Same with `get-status-fn`